lib/pghero/methods/citus.rb

```diff
@@ -0,0 +1,17 @@
+module PgHero
+  module Methods
+    module Citus
+      # It will only cache citus_enabled if it is true once.
+      def citus_enabled?
+        @citus_enabled ||= citus_readable?
+      end
+
+      def citus_readable?
+        select_all("SELECT 1 FROM pg_dist_partition LIMIT 1")
```

Comment:
I think it would be better if we use something like;

```sql
SELECT EXISTS(SELECT * FROM pg_available_extensions() WHERE name = 'citus');
```

That would probably save us from except'on handling too.

Reply:
pg_extension would be appropriate because citus may be available but not installed yet. I have used this way to make sure pg_dist_partition is there. I wonder, is there any possibility that citus extension has been created but somehow pg_dist_partition is not there?

Reply:
Right, `pg_extension` is a better choice. I think we should not rely on the existence of `pg_dist_partition`, because;

- Although it is rare, we can change catalog tables in the future.
- We have other catalog tables as well, randomly picking one feels awkward
- Using exception handling for checking availability is bad pattern.

Reply:
I agree on the third point. Regarding the first two, I am not randomly picking one. Maybe I should add a comment that explains that. `pg_dist_partition` is a crucial catalog to make PgHero Citus-compatible, and almost every method uses it. Changing this catalog table would require changing the methods as well anyway. If somehow `pg_dist_partition` is not there, the respective methods would error out. Rather than checking for its existence in all the methods, I have chosen to check it in `citus_enabled`. However, if it is extremely rare that `citus` extension is created and `pg_dist_partition` not, then we should use `pg_extension` because, as you said, it saves from exception handling.